### PR TITLE
Dilpreet | Test | User Panel Bug

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/add-company/add-company.component.html [hunk 1]
```diff
@@ -176,6 +176,9 @@
                             <button mat-stroked-button color="primary" (click)="nextStepForm()">
                                 {{ commonLocaleData?.app_next }}
                             </button>
+                            <button mat-stroked-button color="primary" *ngIf="isCreateBySubscription" (click)="back()">
```
Comment (nisha-walkover) [inline apps/web-giddh/src/app/add-company/add-company.component.html]: add aria-label in all button tag for accessibility

File: apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.html [hunk 2]
```diff
@@ -43,43 +23,52 @@ <h2 class="no-data">
...
-                                                >
-                                                    <mat-button-toggle value="MONTHLY">{{
-                                                        commonLocaleData?.app_duration?.monthly
-                                                    }}</mat-button-toggle>
-                                                    <mat-button-toggle value="YEARLY">{{
-                                                        commonLocaleData?.app_duration?.yearly
-                                                    }}</mat-button-toggle>
-                                                </mat-button-toggle-group>
-                                            </div>
+                                        <!-- <div class="form-group col-md-2" *ngIf="isNewUserLoggedIn"> -->
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.html:37]: Commented code 
??

File: apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts [hunk 3]
```diff
@@ -201,10 +205,28 @@ export class BuyPlanComponent implements OnInit, OnDestroy {
             }
         });
 
+        this.updatePlanSuccess$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts:209]: suggestion

if (response && response?.duration) {

File: apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts [hunk 4]
```diff
@@ -201,10 +205,28 @@ export class BuyPlanComponent implements OnInit, OnDestroy {
...
+            }
+        });
+
+
         this.updateSubscriptionPaymentIsSuccess$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
             if (response) {
                 this.isLoading = false;
-                this.router.navigate(['/pages/new-company/' + this.subscriptionId]);
+                if (this.isChangePlan) {
+                    this.router.navigate(['/pages/subscription'])
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts]: this.router.navigate(['/pages/subscription']);

File: apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts [hunk 5]
```diff
@@ -201,10 +205,28 @@ export class BuyPlanComponent implements OnInit, OnDestroy {
...
+                } else {
+                    this.router.navigate(['/pages/new-company/' + this.subscriptionId])
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts:228]: this.router.navigate(['/pages/new-company/' + this.subscriptionId]);

suggestion
this.router.navigate([`/pages/new-company/${this.subscriptionId}`]);

File: apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts [hunk 6]
```diff
@@ -721,50 +710,82 @@ export class BuyPlanComponent implements OnInit, OnDestroy {
...
-                }
-
+                this.setPlans();
             } else {
                 this.inputData = [];
             }
         });
     }
 
+    private setPlans(): void {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts:736]: documentation

File: apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts [hunk 7]
```diff
@@ -14,7 +14,7 @@ export interface BuyPlanState {
     createSubscriptionSuccess: boolean;
     createSubscriptionResponse: any;
     createSubscriptionInProgress: boolean;
-    updatePlanSuccess: boolean;
+    updatePlanSuccess: any;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts:17]: Why use any ??

File: apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts [hunk 8]
```diff
@@ -152,18 +152,18 @@ export class BuyPlanComponentStore extends ComponentStore<BuyPlanState> implemen
                     tapResponse(
                         (res: BaseResponse<any, any>) => {
                             if (res?.status === 'success') {
-                                this.toasterService.showSnackBar('success', 'Update Plan Successfully');
+                                this.toasterService.showSnackBar('success', 'Plan update Successfully');
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/subscription/buy-plan/utility/buy-plan.store.ts:155]: 'Plan update Successfully'

translate this message

File: apps/web-giddh/src/app/subscription/view-subscription/view-subscription.component.ts [hunk 9]
```diff
@@ -168,7 +168,7 @@ export class ViewSubscriptionComponent implements OnInit, OnDestroy {
      * @memberof ViewSubscriptionComponent
      */
     public changeBilling(): void {
-        this.router.navigate(['/pages/subscription/change-billing/' + this.viewSubscriptionData?.billingAccount?.uniqueName]);
+        this.router.navigate(['/pages/subscription/change-billing/' + this.viewSubscriptionData?.billingAccountUniqueName]);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/subscription/view-subscription/view-subscription.component.ts]: Suggestion

this.router.navigate([`/pages/subscription/change-billing/${this.viewSubscriptionData?.billingAccountUniqueName}`]);

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

**Review by dvCodeWorld — APPROVED** [review]

File: apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.html [hunk 10]
```diff
@@ -89,527 +78,523 @@ <h2 class="font-16 strong">{{ localeData?.choose_plans }}</h2>
...
-                                                    >
-                                                        <span>{{ plan?.billsAllowed }}</span>
-                                                    </div>
-                                                </ng-container>
+                                            </ng-container>
+                                        </div>
+                                        <div class="grid-list-div list-grid">
+                                            <div class="grid-title-left">
+                                                <span class="inner-name mat-table-sticky">
+                                                    <span>Companies Allowed </span>
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.html]: translation?

File: apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts [hunk 11]
```diff
@@ -721,50 +710,88 @@ export class BuyPlanComponent implements OnInit, OnDestroy {
...
+        this.setFinalAmount();
+    }
+
+    /**
+     * This will be use for set final amount
+     *
+     * @memberof BuyPlanComponent
+     */
+    public setFinalAmount(): void {
+        if (this.secondStepForm?.get('country')?.value?.value?.toLowerCase() === 'in' && !this.promoCodeResponse?.length) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts:757]:
```suggestion
        if (this.secondStepForm?.get('country')?.value?.toLowerCase() === 'in' && !this.promoCodeResponse?.length) {
```

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**